src/gdext/surface/properties.nim

```diff
@@ -65,13 +84,13 @@ macro register_property*[T: SomeUserClass; P: SomeProperty](
       usage: set[PropertyUsageFlags] = PropertyUsageFlags.propertyUsageDefault;
     ): untyped =
   result = newStmtList()
-
   let
+    typname = $typ.getTypeInst[1].toStrLit
     gettersym =
-      if getter.kind == nnkLambda: genSym(nskProc, "get_" & $name)
+      if getter.kind == nnkLambda: genSym(nskProc, "get_" & typname & "_" & $name)
       else: getter
     settersym =
-      if setter.kind == nnkLambda: genSym(nskProc, "set_" & $name)
+      if setter.kind == nnkLambda: genSym(nskProc, "set_" & typname & "_" & $name)
```

Comment:
No changes are required in this area. It is true that Godot does not allow method overloading, but unlike Nim, it is a pure OOP language, so all functions belong to classes. Since the namespace is different, no batting occurs.